sbol_utilities/default_components.py

```diff
@@ -0,0 +1,148 @@
+
+def deg_tag(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a degradation tag and a sequence Components with a given name, description, and sequence.
+    """
+    deg_tag_seq = sbol3.Sequence(f'{name}_seq')
+    deg_tag_seq.elements= sequence
+    deg_tag_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    deg_tag = sbol3.Component(name, sbol3.SBO_DNA)
+    deg_tag.name = name
+    deg_tag.description = description
+    deg_tag.roles. append(tyto.SO.protein_stability_element)
+    deg_tag.sequences.append(deg_tag_seq)
+    return tuple([deg_tag, deg_tag_seq])
+
+def plasmid_vector(name: str, description: str, sequence: str)-> Tuple[sbol3.Component, sbol3.Sequence]:
+    """
+    Creates a plasmid vector or backbone and a sequence Components with a given name, description, and sequence.
+    """
+    plasmid_vector_seq = sbol3.Sequence(f'{name}_seq')
+    plasmid_vector_seq.elements= sequence
+    plasmid_vector_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    plasmid_vector = sbol3.Component(name, sbol3.SBO_DNA)
+    plasmid_vector.name = name
+    plasmid_vector.description = description
+    plasmid_vector.roles. append(tyto.SO.plasmid_vector)
+    plasmid_vector.sequences.append(plasmid_vector_seq)
+    return tuple([plasmid_vector, plasmid_vector_seq])
+
+def engineered_region(name: str, description: str, sub_components: List[sbol3.SubComponent])-> sbol3.Component:
+    """
+    Creates an engineered region Componentswith a given name, description, and sub components.
+    """
+    #sequence = blunt or assembly?
+    #engineered_region_seq = sbol3.Sequence(f'{name}_seq')
+    #engineered_region_seq.elements= sequence
+    #engineered_region_seq.encoding = 'https://identifiers.org/edam:format_1207'
+
+    engineered_region = sbol3.Component(name, sbol3.SBO_DNA)
+    engineered_region.name = name
+    engineered_region.description = description
+    engineered_region.roles.append(sbol3.SO_ENGINEERED_REGION)
+    #create subcomponents if Components are passed  
+    for x in sub_components:
+        engineered_region.features.append(x)
+    #fix order
+    if len(engineered_region.features) > 1:
+            for i in range(len(engineered_region.features)-1):
+                engineered_region.constraints = [sbol3.Constraint(sbol3.SBOL_PRECEDES, engineered_region.features[i], engineered_region.features[i+1])]
+    else: pass
+    #engineered_region.sequences.append(engineered_region_seq)
+    return engineered_region # , engineered_region_seq
+
+def media(name: str, description: str)-> sbol3.Component:
+    """
+    Creates a media Component with a given name and description.
+    """
+    media = sbol3.Component(name, sbol3.SBO_FUNCTIONAL_ENTITY)
+    media.name = name
+    media.description = description
+    media.roles.append(tyto.NCIT.Media)
+    return media
+
+def strain(name: str, description: str)-> sbol3.Component:
```

Comment:
Don't you want to add any other information about a strain?

Reply:
hmmm I dont know what could be the best. Maybe biocyc, but I dont know if produces stable URIs, that is an option. In Newcastle are developing a kind of GitHub for strains and engineered organisms, I will look for more information, I dont remember the name right now.

Reply:
Take a look at Section 7.10 in the specification, which has the current best practice recommendations (which were developed in part by Newcastle).

Reply:
The SBOL Specification 3.0.1 section 7.10 RECOMMEND to the type “cell” from the Gene Ontology (GO:0005623). Looked for this term and it is [obsolete ](https://www.ebi.ac.uk/QuickGO/term/GO:0005623). I've raised an issue on SBOL specification to address it (https://github.com/SynBioDex/SBOL-specification/issues/474). The constructor now provides strain and the kwargs can handle any other information.

This is how looks the function:
```
def strain(identity: str, **kwargs)-> sbol3.Component:
    """
    Creates a strain Component of type functional entity.

    :param identity: The identity of the Component.
    :param **kwargs: Keyword arguments of any other Component attribute.
    :return: A Component object.
    """
    strain =  sbol3.Component(identity, sbol3.SBO_FUNCTIONAL_ENTITY, **kwargs)
    strain.roles.append(tyto.NCIT.Strain)
    return strain
```